ui/ui-components/client/client.ts

```diff
@@ -83,92 +89,91 @@ export class Client {
   private serverToken = process.env.SERVER_TOKEN;
   private cache = new ClientCache();
 
-  private static readonly optionDefaults: ClientRequestOptions = {
+  private static readonly defaultOptions: ClientRequestOptions = {
     useCache: true,
-    errorHandler,
+    errorHandler: isBrowser() ? errorHandler : undefined,
+  };
+
+  private static readonly defaultHeaders: AxiosRequestHeaders = {
+    Accept: "application/json",
+    "Content-Type": "application/json",
+    "X-Grouparoo-Client": `${PackageJSON.name}-v${PackageJSON.version}`,
   };
 
   constructor(
     private getRequestContext: () => {
+      type?: NextContextType;
       req?: IncomingMessage;
       res?: ServerResponse;
     } = () => ({})
   ) {}
 
-  private csrfToken = () => {
-    if (globalThis?.localStorage) {
-      return window.localStorage.getItem("session:csrfToken");
-    }
-  };
+  private redirectOnAccessError({ code }: { code: string }): boolean {
+    if (isBrowser()) {
+      switch (code) {
+        case "AUTHENTICATION_ERROR":
+          if (window.location.pathname !== "/session/sign-in") {
+            window.location.href = `/session/sign-in?nextPage=${window.location.pathname}`;
+          }
+          return true;
 
-  public request = async <Response = any>(
-    verb: Method = "get",
-    path: string,
-    data: AxiosRequestConfig["data"] = {},
-    options: Partial<ClientRequestOptions> = {}
-  ): Promise<Response> => {
-    options = { ...Client.optionDefaults, ...options };
+        case "NO_TEAMS_ERROR":
+          window.location.href = `/`;
+          return true;
 
-    const headers: AxiosRequestHeaders = {
-      Accept: "application/json",
-      "Content-Type": "application/json",
-      "X-Grouparoo-Client": `${PackageJSON.name}-v${PackageJSON.version}`,
-    };
+        case "AUTHORIZATION_ERROR":
+        default:
+          // ok, it will be rendered on the page
+          break;
+      }
 
-    const { req } = this.getRequestContext();
-    if (req?.headers?.cookie) {
-      headers["X-GROUPAROO-SERVER-TOKEN"] = this.serverToken;
-      headers["cookie"] = req?.headers?.cookie;
-      options.useCache = false; // do not ever cache responses on the server
+      return false;
     }
 
-    const config: AxiosRequestConfig = {
-      params: null,
-      data: null,
-      url: `${this.webUrl}/api/${this.apiVersion}${path}`, // path comes with a leading "/"
-      withCredentials: true,
-      method: verb.toLowerCase() as Method,
-      headers,
-    };
-
-    data.csrfToken = this.csrfToken();
+    const { type, req, res } = this.getRequestContext();
+    if (!req || !res || (type !== "AppContext" && type !== "NextPageContext")) {
```

Comment:
I don't anticipate this value changing, why not just
`type === "ServerSidePropsContext"`